src/components/sidenav/sidenav.js

```diff
@@ -257,9 +266,30 @@ function SidenavDirective($mdMedia, $mdUtil, $mdConstant, $mdTheming, $animate,
     scope.$watch('isOpen', updateIsOpen);
 
 
+
     // Publish special accessor for the Controller instance
     sidenavCtrl.$toggleOpen = toggleOpen;
 
+    function revalidateVisibility() {
+      var lastValue = scope.isOpen;
+      if (isHidden(element, true, true) && scope.isOpen == true) scope.isOpen = false;
+      else if (!isHidden(element, true, true) && scope.isOpen == false) scope.isOpen = true;
```

Comment:
Brackets, `scope.isOpen == true` is the same as `scope.isOpen` and ternary operator